rules/manager.go

```diff
@@ -907,6 +907,34 @@ func (m *Manager) AlertingRules() []*AlertingRule {
 	return alerts
 }
 
+// AlertingRules returns the list of the manager's alerting rules organised by group.
+func (m *Manager) AlertingRulesbyGroup() *AlertingGroups {
+	m.mtx.RLock()
+	defer m.mtx.RUnlock()
+
+	alerts := &AlertingGroups{}
+	for _, g := range m.groups {
+		group := &AlertingGroup{Name: g.name}
+		for _, rule := range g.rules {
+			if alertingRule, ok := rule.(*AlertingRule); ok {
+				group.Rules = append(group.Rules, alertingRule)
+			}
+		}
+		alerts.Groups = append(alerts.Groups, group)
+	}
+	return alerts
+}
+
+type AlertingGroups struct {
+	Groups []*AlertingGroup
+}
+
+type AlertingGroup struct {
```

Comment:
Hmm, I think it might be better to just return the existing `rule.Group` structs here and add `Name()` and `AlertingRules()` methods on that type directly.